zerver/lib/actions.py

```diff
@@ -568,7 +568,9 @@ def do_create_user(email: str, password: Optional[str], realm: Realm, full_name:
                    newsletter_data: Optional[Dict[str, str]]=None,
                    default_stream_groups: Sequence[DefaultStreamGroup]=[],
                    source_profile: Optional[UserProfile]=None,
-                   realm_creation: bool=False) -> UserProfile:
+                   realm_creation: bool=False,
+                   acting_user: Optional[UserProfile]=None,
+                   user_registered_itself: bool=False) -> UserProfile:
```

Comment:
I suspect you don't need `user_registered_itself`; is there a case where it should not be implied by `acting_user=None`.

Reply:
@timabbott  I think we do want to have acting_user to be the modified_user itself here. Couldn't found a way how we can do that by not passing the user_registered_itself argument since we can't pass the user_profile here.

Reply:
Oh, my suggestion was `acting_user=None` could be the conditional for "mark this user as self-created", without a separate flag needed.  I think that `acting_user=None` will be what we pass for that parameter exactly in the cases where we'd pass `user_registered_itself`.

Reply:
@timabbott Okay just that we pass none for management commands and in backend.py. So won't we have to differentiate this with them?